compiler/rustc_target/src/spec/i686_apple_darwin.rs

```diff
@@ -1,7 +1,7 @@
 use crate::spec::{FramePointer, LinkerFlavor, StackProbeType, Target, TargetOptions};
 
 pub fn target() -> Target {
-    let mut base = super::apple_base::opts("macos");
+    let mut base = super::apple_base::opts("macos", "i386", None);
```

Comment:
Arch is "i386" here, but "i686" below.

Reply:
ld64 only seems to understand i386 here.
Apple's clang accepts i686 and tries to pass it to ld64 but this end up failing:

```
  = note: ld: unknown/unsupported architecture name for: -arch i686
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Before my changes the arch wasn't passed for this specific target

Reply:
Could you add a comment explaining this difference?